geopandas/tools/tests/test_sjoin.py

```diff
@@ -384,6 +391,37 @@ def test_sjoin_dwithin(
         joined = sjoin(left, right, how=how, predicate="dwithin", distance=distance)
         assert_frame_equal(expected_gdf.sort_index(), joined.sort_index())
 
+    # GH3239
+    @pytest.mark.parametrize(
+        "predicate",
+        [
+            "contains",
+            "contains_properly",
+            "covered_by",
+            "covers",
+            "crosses",
+            "intersects",
+            "touches",
+            "within",
+        ],
+    )
+    def test_sjoin_left_order(self, predicate):
```

Comment:
```suggestion
    def test_sjoin_left_order(self, predicate):
        # a set of points in random order -> that order should be preserved
        # with a left join
```

(so we don't accidentally change it to a simple order later)